src/Widget/WidgetTrait.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Widget;
+
+use Yii\Extension\Bulma\Asset\BulmaAsset;
+use Yiisoft\Assets\AssetManager;
+use Yiisoft\Json\Json;
+
+trait WidgetTrait
+{
+    private ?AssetManager $assetManager = null;
+    private array $clientOptions = [];
+    private array $clientEvents = [];
+    private bool $enableClientOptions = false;
+    private ?object $webView = null;
+
+    /**
+     * Registers a specific Bulma plugin and the related events.
+     *
+     * @param string $name the name of the Bulma plugin
+     * @param array $options
+     *
+     * @return void
+     */
+    protected function registerPlugin(string $name, array $options = []): void
+    {
+        $id = $options['id'];
+
+        $new = clone $this;
+
+        if ($new->assetManager !== null) {
+            $new->assetManager->register([
+                BulmaAsset::class
+            ]);
+        }
+
+        if ($new->webView !== null) {
+            $new->webView->setCssFiles($new->assetManager->getCssFiles());
+            $new->webView->setJsFiles($new->assetManager->getJsFiles());
+        }
+
+        if ($new->enableClientOptions !== false) {
+            $optionsString = Json::htmlEncode($new->clientOptions);
+            $js = "jQuery('#$id').$name($optionsString);";
+
+            if ($new->webView !== null) {
+                $new->webView->registerJs($js);
+            }
+        }
+
+        $new->registerClientEvents($id);
+    }
+
+    /**
+     * Registers JS event handlers that are listed in {@see clientEvents}.
+     *
+     * @param string $id
+     */
+    protected function registerClientEvents(string $id): void
+    {
+        $new = clone $this;
+
+        if ($new->clientEvents) {
+            $js = [];
+
+            foreach ($new->clientEvents as $event => $handler) {
+                $js[] = "jQuery('#$id').on('$event', $handler);";
+            }
+
+            if ($new->webView !== null) {
+                $new->webView->registerJs(implode("\n", $js));
+            }
+        }
+    }
+
+    public function assetManager(AssetManager $value): self
+    {
+        $new = clone $this;
+        $new->assetManager = $value;
+        return $new;
+    }
+
+    /**
+     * The event handlers for the underlying Bootstrap JS plugin.
+     *
+     * Please refer to the corresponding Bootstrap plugin Web page for possible events.
```

Comment:
That's not about bootstrap. We're using bulma, aren't we?

Reply:
done.